core/integration/blueprint_test.go

```diff
@@ -0,0 +1,270 @@
+			// Create blueprint module
+			WithNewFile("/tmp/blueprint-init-sdk-test/main.go", blueprintSrc).
+			WithWorkdir("/tmp/blueprint-init-sdk-test").
+			With(daggerExec("init", "--name=blueprint-init-sdk-test", "--sdk=go", ".")).
+			// Try to init with both SDK and blueprint - should fail
+			WithWorkdir("/tmp/target-init-sdk-test")
+
+		// Try to init with both SDK and blueprint - should fail
+		_, err := modGen.
+			With(daggerExec("init", "--name=target-init-sdk-test", "--sdk=go", "--blueprint=../blueprint-init-sdk-test", ".")).
+			Stdout(ctx)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "cannot specify both --sdk and --blueprint")
+	})
+}
+
+func (BlueprintSuite) TestBlueprintContextSeparation(ctx context.Context, t *testctx.T) {
+	c := connect(ctx, t)
+
+	// Blueprint that tests context separation
+	blueprintSrc := `package main
+
+import (
+	"context"
+	"dagger/blueprint-context-test/internal/dagger"
+)
+
+type Blueprint struct{}
+
+// This should read blueprint's own config file
+func (m *Blueprint) ReadBlueprintConfig(ctx context.Context) (string, error) {
+	dir := dag.CurrentModule().Source().ContextDirectory()
+	content, err := dir.File("blueprint-config.txt").Contents(ctx)
+	if err != nil {
+		return "", err
+	}
+	return content, nil
+}
+
+// This should return the target module's name, not blueprint's
+func (m *Blueprint) GetActualModuleName(ctx context.Context) (string, error) {
+	return dag.CurrentModule().Name(), nil
+}
+`
+
+	// Test context separation
+	t.Run("context separation", func(ctx context.Context, t *testctx.T) {
+		modGen := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/tmp").
+			// Create blueprint module with its own config
+			WithNewFile("/tmp/blueprint-context-test/main.go", blueprintSrc).
+			WithNewFile("/tmp/blueprint-context-test/blueprint-config.txt", "blueprint-specific-config").
+			WithWorkdir("/tmp/blueprint-context-test").
+			With(daggerExec("init", "--name=blueprint-context-test", "--sdk=go", ".")).
+			// Create target module with its own data
+			WithWorkdir("/tmp/target-context-test").
+			WithNewFile("/tmp/target-context-test/target-data.txt", "target-specific-data").
+			With(daggerExec("init", "--name=target-context-test", "--blueprint=../blueprint-context-test", "."))
+
+		// Test 1: Blueprint should be able to read its own config
+		out, err := modGen.
+			With(daggerExec("call", "read-blueprint-config")).
+			Stdout(ctx)
+		require.NoError(t, err)
+		require.Contains(t, out, "blueprint-specific-config")
+
+		// Test 2: Module name should be target's name, not blueprint's
+		out, err = modGen.
+			With(daggerExec("call", "get-actual-module-name")).
+			Stdout(ctx)
+		require.NoError(t, err)
+		require.Contains(t, out, "target-context-test")
+		require.NotContains(t, out, "blueprint-context-test")
+	})
+}
+
+func (BlueprintSuite) TestBlueprintConfigurationPersistence(ctx context.Context, t *testctx.T) {
```

Comment:
Tests LGTM (haven't looked too closely), but could you add a test for a blueprint with a dependency? Just feels like an edge case that might *potentially* break in the future if we don't have a test :pray: